web_src/js/features/repo-diff.js

```diff
@@ -119,37 +119,29 @@ function onShowMoreFiles() {
   countAndUpdateViewedFiles();
 }
 
-export function doLoadMoreFiles(link, diffEnd, callback) {
-  const url = `${link}?skip-to=${diffEnd}&file-only=true`;
-  loadMoreFiles(url, callback);
-}
-
-function loadMoreFiles(url, callback) {
+export function loadMoreFiles(url) {
   const $target = $('a#diff-show-more-files');
-  if ($target.hasClass('disabled')) {
-    callback();
+  if ($target.hasClass('disabled') || pageData.diffFileInfo.isLoadingNewData) {
```

Comment:
Why we need to check both `isLoadingNewData` and `disabled ` class here? A bit confused if a single check for `isLoadingNewData` works.

Reply:
No particular reason. But disabled means traditional MVC while isXxx is for reactive MVVM. It's also right to do so And remind readers that this logic is shared by MVC and MVVM